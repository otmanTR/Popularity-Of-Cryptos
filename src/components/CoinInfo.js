import PropTypes from 'prop-types';
import React from 'react';

const CoinInfo = ({
  image,
  price,
  name,
  symbol,
  marketCap,
  marketCapRank,
  totalVolume,
  high24h,
  low24h,
  priceChange24h,
  marketCapChange24h,
  totalSupply,
}) => (
  <>
    <div>
      <div>
        <img className="coinImage" src={image} alt={image} />
      </div>
      <div className="infoText">
        <p>Crypto Coin Info:</p>
      </div>
      <ul className="detailUl">
        <li className="detail">
          Name:
          {name}
        </li>
        <li className="detail">
          Symbol:
          {symbol}
        </li>
        <li className="detail">
          Price:
          {price}
          $
        </li>
        <li className="detail">
          Market Cap:
          {marketCap}
          $
        </li>
        <li className="detail">
          Market Cap Rank:
          {marketCapRank}
          st.
        </li>
        <li className="detail">
          Total Volume:
          {totalVolume}
          $
        </li>
        <li className="detail">
          Highest 24h:
          {high24h}
          $
        </li>
        <li className="detail">
          Lowest 24h
          {low24h}
          $
        </li>
        <li className="detail">
          Price Change 24h:
          {priceChange24h}
          $
        </li>
        <li className="detail">
          Market Cap Change:
          {marketCapChange24h}
          $
        </li>
        <li className="detail">
          Total Supply:
          {totalSupply}
        </li>
      </ul>
    </div>
  </>
);

CoinInfo.propTypes = {
  marketCap: PropTypes.string.isRequired,
  marketCapRank: PropTypes.string.isRequired,
  totalVolume: PropTypes.string.isRequired,
  high24h: PropTypes.string.isRequired,
  low24h: PropTypes.string.isRequired,
  priceChange24h: PropTypes.string.isRequired,
  marketCapChange24h: PropTypes.string.isRequired,
  totalSupply: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  price: PropTypes.string.isRequired,
  symbol: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
};

export default CoinInfo;
