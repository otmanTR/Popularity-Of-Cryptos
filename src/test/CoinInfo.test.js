import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import CoinInfo from '../components/CoinInfo';

describe('CoinInfo Component', () => {
  test('component match with the snapshot', () => {
    const tree = renderer.create(<BrowserRouter><CoinInfo /></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
