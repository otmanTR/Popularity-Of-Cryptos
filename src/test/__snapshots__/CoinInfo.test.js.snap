// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoinInfo Component component match with the snapshot 1`] = `
<div
  className="coinDetails"
>
  <div>
    <img
      className="coinImage"
    />
  </div>
  <div
    className="infoText"
  >
    <h2>
      Crypto Coin Info:
    </h2>
  </div>
  <ul
    className="detailUl"
  >
    <li
      className="detail"
    >
      Name:
       
       
    </li>
    <li
      className="detail"
    >
      Symbol:
       
       
    </li>
    <li
      className="detail"
    >
      Price:
       
       
      $
    </li>
    <li
      className="detail"
    >
      Market Cap:
       
       
      $
    </li>
    <li
      className="detail"
    >
      Market Cap Rank:
       
       
      st.
    </li>
    <li
      className="detail"
    >
      Total Volume:
       
       
      $
    </li>
    <li
      className="detail"
    >
      Highest 24h:
       
       
      $
    </li>
    <li
      className="detail"
    >
      Lowest 24h:
       
       
      $
    </li>
    <li
      className="detail"
    >
      Price Change 24h:
       
       
      $
    </li>
    <li
      className="detail"
    >
      Market Cap Change:
       
       
      $
    </li>
    <li
      className="detail"
    >
      Total Supply:
       
       
      piece
    </li>
  </ul>
</div>
`;
